crates/win-api-wrappers/src/handle.rs

```diff
@@ -19,15 +19,44 @@ unsafe impl Send for Handle {}
 unsafe impl Sync for Handle {}
 
 impl Handle {
-    pub fn new(handle: HANDLE, owned: bool) -> Self {
+    // Wraps a Windows [`HANDLE`].
+    //
+    // # Safety
+    //
+    // When `owned` is `true`:
+    //
+    // - `handle` is a valid handle to an open object.
+    // - `handle` is not a pseudohandle.
+    // - The caller is actually responsible for freeing the `HANDLE` when the value goes out of scope.
+    //
+    // When `owned` is `false`: no outstanding precondition.
+    pub unsafe fn new(handle: HANDLE, owned: bool) -> Self {
         Self { raw: handle, owned }
     }
 
+    // Wraps an owned Windows [`HANDLE`].
+    //
+    // # Safety
+    //
+    // - `handle` is a valid handle to an open object.
+    // - `handle` is not a pseudohandle.
+    // - The caller is actually responsible for freeing the `HANDLE` when the value goes out of scope.
+    pub unsafe fn new_owned(handle: HANDLE) -> Self {
+        // SAFETY: Same preconditions as the called function.
+        unsafe { Self::new(handle, true) }
+    }
+
+    // Wraps a borrowed Windows [`HANDLE`].
+    pub fn new_borrowed(handle: HANDLE) -> Self {
+        // SAFETY: It’s safe to wrap a non-owning Handle as we’ll not call `CloseHandle` on it.
+        unsafe { Self::new(handle, false) }
+    }
```

Comment:
When we know that the handle is not owned in a static way, we can use the safe function `new_borrowed` which never causes any UB.